Adds contributing guidelines & refreshes documentation (#307)
AFM-SPM/TopoStats: docs/configuration.md

# Configuration

Configuration for TopoStats is done using a [YAML](https://yaml.org/) configuration that is specified on the command line when
invoking. A default configuration file is provided in the TopoStats repository at
[`topostats/default_config.yaml`](https://github.com/AFM-SPM/TopoStats/blob/dev/topostats/default_config.yaml). The file
contains comments indicating valid values for many of the fields. If no configuration file is provided this default
configuration is loaded automatically and used.

You can make a copy of the `default_config.yaml` and modify it for your own use. Once saved you can run TopoStats with
this configuration file as shown below.

``` bash
run_topostats --config my_config.yaml
```

On completion a copy of the configuration that was used is written to the output directory.

> ## Fields

Reviewer: I don't think that the io channel field has been added

Aside from the comments in YAML file itself the fields are described below.

| Section      | Sub-Section                    | Data Type  | Default        | Description                                                                                                                         |
|:-------------|:-------------------------------|:-----------|:---------------|:------------------------------------------------------------------------------------------------------------------------------------|

> | `base_dir`   |                                | string     | `./`           |                                                                                                                                     |

Reviewer: ```suggestion
| `base_dir`   |                                | string     | `./`           | Directory to recursively search for files within.                                                                                                                                   |
```

| `output_dir` |                                | string     | `./output`     | Directory that output should be saved to.                                                                                           |

> | `warnings`   |                                | string     | `ignore`       | Turns of warnings being shown.                                                                                                      |

Reviewer: ```suggestion
| `warnings`   |                                | string     | `ignore`       | Turns off warnings being shown.                                                                                                      |
```

> | `cores`      |                                | integer    | `4`            |                                                                                                                                     |

Reviewer: ```suggestion
| `cores`      |                                | integer    | `4`            | Number of cores to run parallel processes on.                                                                                                                                    |
```

| `quiet`      |                                | false      |                |                                                                                                                                     |

> | `file_ext`   |                                | string     | `.spm`         |                                                                                                                                     |

Reviewer: ```suggestion
| `file_ext`   |                                | string     | `.spm`         | File extensions to search for.                                                                                                                                 |
```

|              | `save_cropped_grains`          | boolean    | `true`         | Options : true, false                                                                                                               |
| `dnatracing` | `run`                          | boolean    | `true`         | Whether to run DNA Tracing.  Options : true, false                                                                                  |
| `plotting`   | `run`                          | boolean    | `true`         | Whether to run plotting. Options : `true`, `false`                                                                                  |
|              | `save_format`                  | string     | `png`          | Format to save images in, see [matplotlib.pyplot.savefig](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.savefig.html) |
|              | `image_set`                    | string     | `all`          | Which images to plot. Options : `all`, `core`                                                                                       |

> |              | `zrange`                       | list       | `[0, 3]`       | # low and high height range for core images (can take [null, null])                                                                 |

Reviewer: ```suggestion
|              | `zrange`                       | list       | `[0, 3]`       | Low and high height range for core images (can take [null, null])                                                                 |
```

|              | `colorbar`                     | boolean    | `true`         | Whether to include the colorbar scale in plots. Options `true`, `false`                                                             |

> |              | `axes`                         | boolean    | `true`         | # Options : true, false (due to off being a bool when parsed)                                                                       |

Reviewer: ```suggestion
|              | `axes`                         | boolean    | `true`         | Wether to include the axes in the produced plots.                                                                       |
```
